### PR TITLE
Update to target v0.1 API

### PR DESCRIPTION
**Pull request by co9olguy**
Addresses #23

### CODE REVIEW

File: pennylane_ionq/api_client.py [hunk 1]
```diff
@@ -482,7 +482,6 @@ class JobResult(Resource):
     """
 
     SUPPORTED_METHODS = ("GET",)
-    PATH = "v0/jobs/{job_id}/data"
```
Comment (co9olguy) [inline pennylane_ionq/api_client.py:485]: These parts appear to be legacy code that is never used in this plugin's workflow

File: pennylane_ionq/api_client.py [hunk 2]
```diff
@@ -482,7 +482,6 @@ class JobResult(Resource):
...
 
     def __init__(self, job_id, client=None):
```
Comment (josh146) [inline pennylane_ionq/api_client.py:486]:
```suggestion
    def __init__(self, client=None):
```

Comment (josh146) [reply pennylane_ionq/api_client.py:486]: Although this might necessitate change elsewhere

File: pennylane_ionq/api_client.py [hunk 3]
```diff
@@ -503,7 +501,6 @@ class JobCircuit(Resource):
     """
 
     SUPPORTED_METHODS = ("GET",)
-    PATH = "v0/jobs/{job_id}/circuit"
 
     def __init__(self, job_id, client=None):
```
Comment (josh146) [inline pennylane_ionq/api_client.py:505]:
```suggestion
    def __init__(self, client=None):
```

**Review by josh146 — APPROVED** [review]
Looks good on my end @co9olguy, left a small comment regarding the codefactor complaints -- not sure if it's worth fixing or not.

Could you also update the changelog with a link back to this PR? Thanks!

**Comment by amilstead** [comment]
Thanks so much @josh146!!

**Pull request merged**